tests/ui/octal_escapes.rs

```diff
@@ -0,0 +1,12 @@
+#![warn(clippy::octal_escapes)]
+
+fn main() {
+    let _bad1 = "\033[0m";
+    let _bad2 = b"\033[0m";
+    let _bad3 = "\\\033[0m";
+    let _bad4 = "\01234567";
+    let _bad5 = "\0\03";
```

Comment:
Do you want clippy to catch both instances then? For simplicity the lint currently only recognizes the first instance per literal.

Reply:
Detecting all instance in one lint message would be ideal, however not required. We can also create an issue to have this as a future enhancement. :upside_down_face: 

One test for single number octal escapes would also be great like this:

```rs
let _bad7 = "EvenMoreText-\01\02-ShortEscapes"
```

Reply:
> Detecting all instance in one lint message would be ideal, however not required.

I already did that anyway since after the rewrite of the iteration code it was basically just removing a "break" :fireworks: